lib/active_record/pt_osc_migration.rb

```diff
@@ -2,7 +2,19 @@
 require 'active_record/connection_adapters/mysql_pt_osc_adapter'
 require 'shellwords'
 
+module MigratorExtensions
+  def record_version_state_after_migrating(version)
+    ActiveRecord::Base.logger.debug 'Verifying active connections prior to recording version' if ActiveRecord::Base.logger
+    ActiveRecord::Base.verify_active_connections! # Reconnect to DB if it's gone away while we were migrating.
+    super
+  end
+end
+
 module ActiveRecord
+  class Migrator
+    prepend MigratorExtensions
+  end
```

Comment:
Any reason you can't use [`alias_method_chain`](http://apidock.com/rails/Module/alias_method_chain) to prevent breaking compatibility with Ruby 1.9.3?